CMakeLists.txt

```diff
@@ -78,6 +78,17 @@ include(${PROJECT_SOURCE_DIR}/cmake/dependencies.cmake)
 add_library(umd_common_directories INTERFACE)
 target_include_directories(umd_common_directories INTERFACE ${PROJECT_SOURCE_DIR} ${PROJECT_SOURCE_DIR}/device ${PROJECT_SOURCE_DIR}/third_party/fmt/include)
 
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64" OR CMAKE_SYSTEM_PROCESSOR MATCHES "i386")
```

Comment:
Per offline discussion, include .cpp sources by host arch instead of headers, that way interface to metal will be the same.

The drawback, though, will be that the compiler won't be able to remove function call in that case. Wondering if that might even be a problem... According to github copilot, this shouldn't be a problem.

It could be a perf hit though.
In anycase, these should be in a separate utils repo. Started an issue here: #78

Reply:
These function cannot be inline anymore since the implementation will move to .cpp files. Are we ok with this? 

If we want to keep it inline, we could just guard include headers with defines like 

```
#if defined (__x86_64__)
#include "x86/driver_atomics.h"
#endif

// empty file for device/driver_atomics.h
```

and move the implementation to separate .h files based on the architecture. This way we don't need to change cmake. Next step could be to add dependency on host arch to cmake (or investigate if there already is some dependency) in order to completely remove `device/driver_atomics.h`